nevergrad/optimization/oneshot.py

```diff
@@ -105,74 +124,94 @@ class RandomSearchMaker(base.ParametrizedFamily):
-StupidRandom = RandomSearchMaker(stupid=True).with_name("StupidRandom", register=True)
-CauchyRandomSearch = RandomSearchMaker(cauchy=True).with_name("CauchyRandomSearch", register=True)
+    middle_point=True, scale=.01).set_name("SmallScaleRandomSearchPlusMiddlePoint", register=True)
+StupidRandom = RandomSearchMaker(stupid=True).set_name("StupidRandom", register=True)
+CauchyRandomSearch = RandomSearchMaker(cauchy=True).set_name("CauchyRandomSearch", register=True)
 RandomScaleRandomSearch = RandomSearchMaker(
-    scale="random", middle_point=True).with_name("RandomScaleRandomSearch", register=True)
+    scale="random", middle_point=True).set_name("RandomScaleRandomSearch", register=True)
 RandomScaleRandomSearchPlusMiddlePoint = RandomSearchMaker(
-    scale="random", middle_point=True).with_name("RandomScaleRandomSearchPlusMiddlePoint", register=True)
+    scale="random", middle_point=True).set_name("RandomScaleRandomSearchPlusMiddlePoint", register=True)
 
 
 class _SamplingSearch(OneShotOptimizer):
 
-    def __init__(self, parametrization: IntOrParameter, budget: Optional[int] = None, num_workers: int = 1) -> None:
+    def __init__(
+        self,
+        parametrization: IntOrParameter,
+        budget: Optional[int] = None,
+        num_workers: int = 1,
+        sampler: str = "Halton",
+        scrambled: bool = False,
+        middle_point: bool = False,
+        opposition_mode: Optional[str] = None,
+        cauchy: bool = False,
+        autorescale: bool = False,
+        scale: float = 1.,
+        rescaled: bool = False,
+        recommendation_rule: str = "pessimistic"
+    ) -> None:
         super().__init__(parametrization, budget=budget, num_workers=num_workers)
-        self._parameters = SamplingSearch()  # updated by the parametrized family
         self._sampler_instance: Optional[sequences.Sampler] = None
         self._rescaler: Optional[sequences.Rescaler] = None
         self._opposable_data: Optional[np.ndarray] = None
+        self._sampler = sampler
+        self.opposition_mode = opposition_mode
+        self.middle_point = middle_point
+        self.scrambled = scrambled
+        self.cauchy = cauchy
+        self.autorescale = autorescale
+        self.scale = scale
+        self.rescaled = rescaled
+        self.recommendation_rule = recommendation_rule
 
     @property
     def sampler(self) -> sequences.Sampler:
         if self._sampler_instance is None:
-            budget = None if self.budget is None else self.budget - self._parameters.middle_point
+            budget = None if self.budget is None else self.budget - self.middle_point
             samplers = {"Halton": sequences.HaltonSampler,
                         "Hammersley": sequences.HammersleySampler,
                         "LHS": sequences.LHSSampler,
                         }
-            internal_budget = (budget + 1) // 2 if budget and (self._parameters == "quasi" or self._parameters == "opposite") else budget
```

Comment:
@teytaud there was a bug there, since `self._parameters` is **never** `quasi` or `opposite`, it should have been `self._parameters.opposition_mode`.
I'll correct this in the next PR, I did not want to change all at once.

Reply:
Fix in #523